tinygrad/nn.py

```diff
@@ -30,3 +31,59 @@ def normalize(self, x, mean, var):
     x = (x - mean.reshape(shape=[1, -1, 1, 1])) * self.weight.reshape(shape=[1, -1, 1, 1])
     return x.div(var.add(self.eps).reshape(shape=[1, -1, 1, 1])**0.5) + self.bias.reshape(shape=[1, -1, 1, 1])
 
+class Linear:
+  def __init__(self, in_dim, out_dim, bias=True):
+    self.in_dim = in_dim
+    self.out_dim = out_dim
+    self.use_bias = bias
+    self.weight = Tensor.uniform(in_dim, out_dim)
+    if self.use_bias:
+      self.bias = Tensor.zeros(out_dim)
+
+  def __call__(self, x):
+    B, *dims, D = x.shape
+    x = x.reshape(shape=(B * np.prod(dims).astype(np.int32), D))
```

Comment:
I don't think you need to reshape here, dot should broadcast across the top dimensions.

Reply:
Yes, but I got an error without reshaping the input along the first dimension in the case the input has a rank > 2.